packages/stream-tile-handler/src/tile-document-handler.ts

```diff
@@ -68,11 +69,16 @@ export class TileDocumentHandler implements StreamHandler<TileDocument> {
    * @private
    */
   async _applyGenesis(commit: any, meta: CommitMeta, context: Context): Promise<StreamState> {
-    let payload = commit
-    const isSigned = StreamUtils.isSignedCommit(commit)
+    let payload = StreamUtils.isExpandedCommit(commit) ? commit.commit : commit
+    const isSigned = StreamUtils.isSignedCommit(commit) || (commit as CommitData).envelope
     if (isSigned) {
-      payload = (await context.ipfs.dag.get(commit.link, { timeout: IPFS_GET_TIMEOUT })).value
-      await this._verifySignature(commit, meta, context, payload.header.controllers[0])
+      let envelope = commit
+      if (StreamUtils.isExpandedCommit(commit)) {
```

Comment:
I'm really concerned about all these conditional checks with different behavior depending on whether or not this information was already loaded.  Ideally we would make it so all code paths that lead here do the work to load the inner data, so we can always assume we have an expanded CommitData throughout all the StreamtypeHandlers.

If that really isn't possible today, then we should do this check once, up front at the very beginning of commit application (presumably in `applyCommit`, before calling into the commit-type specific helper functions), and expand it to included the loaded commit data so that all later code can assume it is already done.

Reply:
Agreed on both points.

The reason I didn't do the latter is because we don't always need to do all the same lookups - anchor commit handling, for example, needs to lookup the `proof` while signed commit handling looks up the commit and envelope. I could add this logic to `applyCommit` but then I'd be checking for commit type, doing specific lookups, then passing data into specific handlers also under commit type checks? The code structure looked weird to me.

The former would be ideal but we use `applyCommit` all over the place and while I can look at all places, this would not be a surgical PR anymore.